examples/fscloud/main.tf

```diff
@@ -52,14 +52,14 @@ module "event_notification" {
 }
 
 module "secrets_manager" {
-  source                     = "../../modules/fscloud"
-  resource_group_id          = module.resource_group.resource_group_id
-  region                     = var.region
-  secrets_manager_name       = "${var.prefix}-secrets-manager" #tfsec:ignore:general-secrets-no-plaintext-exposure
-  sm_tags                    = var.resource_tags
-  existing_kms_instance_guid = var.existing_kms_instance_guid
-  kms_key_crn                = var.kms_key_crn
-  existing_en_instance_crn   = module.event_notification.crn
+  source                   = "../../modules/fscloud"
+  resource_group_id        = module.resource_group.resource_group_id
+  region                   = var.region
+  secrets_manager_name     = "${var.prefix}-secrets-manager" #tfsec:ignore:general-secrets-no-plaintext-exposure
+  sm_tags                  = var.resource_tags
+  is_hpcs_key              = true
```

Comment:
You should use a CRN parser to determine if the value of `kms_key_crn` is HPCS as its possible a user could use a key protect key when calling this example. It should not cause any issue with "value not known until apply" since its always going to be a required input to the example and the key will never be created in this example